src/manager/decisions.ts

```diff
@@ -22,24 +22,45 @@ export function resolve(state: State, storageHandler: WrappedStorageHandler, eve
         cookieDomain)
     }
   }
+
+  let freshDecisions = [] as string[]
   try {
-    const freshDecisions = ([] as ParsedParam[]).concat((state.pageUrl && getQueryParameter(state.pageUrl, DECISION_ID_QUERY_PARAM_NAME)) || [])
-    const storedDecisions = storageHandler.findSimilarCookies(DECISION_ID_COOKIE_NAMESPACE)
-    freshDecisions
+    const extractedFreshDecisions = ([] as ParsedParam[]).concat((state.pageUrl && getQueryParameter(state.pageUrl, DECISION_ID_QUERY_PARAM_NAME)) || [])
+    freshDecisions = extractedFreshDecisions
       .map(trim)
       .filter(_nonEmpty)
       .filter(isUUID)
       .filter(_onlyUnique)
-      .forEach(decision => _addDecisionId(decision, state.domain))
-    const allDecisions = freshDecisions
-      .concat(storedDecisions)
-      .map(trim)
+  } catch (e) {
+    eventBus.emitErrorWithMessage('DecisionsResolve', 'Error while extracting new decision ids', e)
+  }
+
+  try {
+    freshDecisions.forEach(decision => _addDecisionId(decision, state.domain))
```

Comment:
it might make sense to move the try into the foreach so we try storing all the decisionIds even if one fails